### PR TITLE
test R['x']['y'] syntax

### PR DESCRIPTION
**Pull request by rfourquet**
By the way, this is not documented. Is someone using this syntax?

### CODE REVIEW

File: src/AbstractAlgebra.jl [hunk 1]
```diff
@@ -604,17 +604,10 @@ export Generic
...
 
-getindex(R::Ring, s::Char) = PolynomialRing(R, s)
-
-getindex(R::NCRing, s::String) = PolynomialRing(R, s)
-
-getindex(R::NCRing, s::Char) = PolynomialRing(R, s)
-
-getindex(R::Tuple{Ring, T}, s::String) where {T} = PolynomialRing(R[1], s)
-
-getindex(R::Tuple{Ring, T}, s::Char) where {T} = PolynomialRing(R[1], s)
```
Comment (rfourquet) [inline src/AbstractAlgebra.jl]: Is it on purpose that no such method is defined for `NCRing`?

Comment (wbhart) [reply src/AbstractAlgebra.jl]: I don't think it is on purpose. You can add it if you wish.

Comment (wbhart) [reply src/AbstractAlgebra.jl]: On the other hand, maybe there is the reason. R[x] is the mathematical notation for a polynomial ring, and I'm not sure off the top of my head whether than includes polynomials over a noncommutative ring. Perhaps @fieker happens to know this off the top of his head.

Comment (rfourquet) [reply src/AbstractAlgebra.jl]: `getindex(R::NCRing, s::String) = PolynomialRing(R, s)` is already defined, so `R["x"]` already works for a noncommutative ring `R`.

File: src/AbstractAlgebra.jl [hunk 2]
```diff
@@ -604,17 +604,10 @@ export Generic
...
+# syntax x = R["x"]["y"]
+getindex(R::Tuple{Ring, Union{PolyElem,NCPolyElem}}, s::Union{String,Char}) = PolynomialRing(R[1], s)
```
Comment (wbhart) [inline src/AbstractAlgebra.jl]: Space after comma.

File: src/AbstractAlgebra.jl (hunk 1 above)
Comment (wbhart) [reply src/AbstractAlgebra.jl]: Ok, then go ahead and make that change and the one below and I will merge this.

Comment (rfourquet) [reply src/AbstractAlgebra.jl]: Ok done, I added tests for the non commutative case too.

**Comment by wbhart** [comment]
Thanks.

**Pull request merged**